static/js/components/GroupSources.jsx

```diff
@@ -0,0 +1,241 @@
+          colSpan={colSpan}
+        >
+          <Grid
+            container
+            direction="row"
+            spacing={3}
+            justify="center"
+            alignItems="center"
+          >
+            <ThumbnailList
+              thumbnails={source.thumbnails}
+              ra={source.ra}
+              dec={source.dec}
+              useGrid={false}
+            />
+            <Grid item>
+              <Suspense fallback={<div>Loading plot...</div>}>
+                <VegaPlot dataUrl={`/api/sources/${source.id}/photometry`} />
+              </Suspense>
+            </Grid>
+          </Grid>
+        </TableCell>
+      </TableRow>
+    );
+  };
+
+  // This is just passed to MUI datatables options -- not meant to be instantiated directly.
+  const renderObjId = (dataIndex) => {
+    const objid = sources[dataIndex].id;
+    return (
+      <a href={`/source/${objid}`} key={`${objid}_objid`}>
+        {objid}
+      </a>
+    );
+  };
+
+  // This is just passed to MUI datatables options -- not meant to be instantiated directly.
+
+  const renderRA = (dataIndex) => {
+    const source = sources[dataIndex];
+    return (
+      <div key={`${source.id}_ra`}>
+        {source.ra}
+        <br />
+        {ra_to_hours(source.ra)}
+      </div>
+    );
+  };
+
+  // This is just passed to MUI datatables options -- not meant to be instantiated directly.
+  const renderDec = (dataIndex) => {
+    const source = sources[dataIndex];
+    return (
+      <div key={`${source.id}_dec`}>
+        {source.dec}
+        <br />
+        {dec_to_hours(source.dec)}
+      </div>
+    );
+  };
+
+  // This is just passed to MUI datatables options -- not meant to be instantiated directly.
+  const renderFinderButton = (dataIndex) => {
+    const source = sources[dataIndex];
+    return (
+      <IconButton size="small" key={`${source.id}_actions`}>
+        <Link href={`/api/sources/${source.id}/finder`}>
+          <PictureAsPdfIcon />
+        </Link>
+      </IconButton>
+    );
+  };
+
+  const columns = [
+    {
+      name: "Source ID",
+      options: {
+        filter: true,
+        customBodyRenderLite: renderObjId,
+      },
+    },
+    {
+      name: "RA",
+      options: {
+        filter: false,
+        customBodyRenderLite: renderRA,
+      },
+    },
+    {
+      name: "Dec",
+      options: {
+        filter: false,
+        customBodyRenderLite: renderDec,
+      },
+    },
+    {
+      name: "Redshift",
+      options: {
+        filter: false,
+      },
+    },
+    {
+      name: "Finder",
+      options: {
+        filter: false,
+        customBodyRenderLite: renderFinderButton,
+      },
+    },
+  ];
+
+  const options = {
+    draggableColumns: { enabled: true },
+    expandableRows: true,
+    renderExpandableRow: renderPullOutRow,
+    selectableRows: "none",
+  };
+
+  const data = sources.map((source) => [
+    source.id,
+    source.ra,
+    source.dec,
+    source.redshift,
```

Comment:
Can we add some more info here? Groups saved to? Classifications? Recent comments?

Reply:
I added groups and comments (do we want to limit the number of displayed comments or just let the user scroll down?). 

I am having some trouble add classifications to sources, so I'm not sure if my addition of classifications to the table is actually working.